subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -201,6 +215,13 @@ static int cracen_signature_prepare_ec_pubkey(struct sitask *t, struct si_sig_pu
 
 			if (PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(psa_get_key_type(attributes))) {
 				pubkey->key.ed25519 = (struct sx_ed25519_pt *)key_buffer;
+				if (message) {
+					status = cracen_signature_set_hashalgo(&pubkey->hashalg,
+									       alg);
+				} else {
+					status = cracen_signature_set_hashalgo_from_digestsz(
```

Comment:
assigning status for nothing :thinking:

Reply:
Actually the entire block is overly complicated for no reason. Changed it back to what it was originally

Reply:
No more `cracen_signature_set_hashalgo*()`?

Reply:
This is in the PURE_EDDSA block, which is not prehashed so does not need it